tests/testlive_comprehensive.py

```diff
@@ -2574,6 +2574,44 @@ def test_blocklists(self):
             for blo in to_delete['bl_organisations']:
                 self.admin_misp_connector.delete_organisation_blocklist(blo)
 
+    def test_event_report(self):
+        event = self.create_simple_event()
+        new_event_report = MISPEventReport()
+        new_event_report.name = "Test Event Report"
+        new_event_report.content = "# Example report markdown"
+        new_event_report.distribution = 5  # Inherit
+        try:
+            event = self.user_misp_connector.add_event(event)
+            new_event_report = self.user_misp_connector.add_event_report(event.id, new_event_report)
+            # The event report should be linked by Event ID
+            self.assertEqual(event.id, new_event_report.event_id)
+
+            event = self.user_misp_connector.get_event(event)
+            # The Event Report should be present on the event
+            self.assertEqual(new_event_report.id, event.event_reports[0].id)
+
+            new_event_report.name = "Updated Event Report"
+            new_event_report.content = "Updated content"
+            new_event_report = self.user_misp_connector.update_event_report(new_event_report)
+            # The event report should be updatable
+            self.assertTrue(new_event_report.name == "Updated Event Report")
+            self.assertTrue(new_event_report.content == "Updated content")
+
+            event_reports = self.user_misp_connector.get_event_reports(event.id)
+            # The event report should be requestable by the Event ID
+            self.assertEqual(new_event_report.id, event_reports[0].id)
+
+            new_event_report = self.user_misp_connector.delete_event_report(new_event_report)
+            # The event report should be soft-deletable
+            self.assertTrue(new_event_report.deleted)
+
+            response = self.user_misp_connector.delete_event_report(new_event_report, True)
```

Comment:
It doesn't seems to return an EventReport there.

Reply:
That'll be the `hard` delete for line 2608-2609. Line 2604-2606 should return an EventReport with the updated `deleted` flag